mllib/src/main/scala/org/apache/spark/mllib/classification/LogisticRegression.scala

```diff
@@ -106,7 +118,6 @@ class LogisticRegressionModel (
 
     // If dataMatrix and weightMatrix have the same dimension, it's binary logistic regression.
     if (numClasses == 2) {
-      require(numFeatures == weightMatrix.size)
```

Comment:
I agree the check is made later but wouldn't you want to fail fast-er, still, if this condition doesn't hold? sorry if I am overlooking something obvious due to your change.